### ISSUE TITLE
"Query is not bootstrapped." When attempting to use Request syntax

### ISSUE THREAD

**Issue by ktomanelli** [byme8/ZeroQL#21]
I'm getting the above error when executing a query using the classes, methods, and schema specified in your example. Looking at the source it seems like this error happens when the GraphQLQueryStore.Executor.TryGetValue returns null. When is this dictionary value being set?

**Comment by byme8** [comment]
It is bootstrapped via source-generated code like that:
<img width="1487" alt="image" src="https://user-images.githubusercontent.com/5927686/206029930-327eef8e-2e4e-4b6f-b377-96429e54c828.png">

It looks like something is wrong with your source generator setup.

**Comment by byme8** [comment]
@ktomanelli  Have you managed to figure out what was wrong?

**Comment by ktomanelli** [comment]
Not yet, diving back into this now. So the dictionary is set upon code generation, as in when running the `dotnet zeroql generate` cli command correct? So if I'm getting this error it means something went wrong during generation?

**Comment by ktomanelli** [comment]
I started a new project and follow the instructions in the readme and am also getting this same error. [Here is a link to the repo](https://github.com/ktomanelli/ZQLQueryIsNotBootStrapped)

**Comment by byme8** [comment]
What IDE do you use?

**Comment by ktomanelli** [comment]
currently using Visual studio for mac

**Comment by byme8** [comment]
It works fine in VS for Mac 17.3.8 for me. What version of do you use?
<img width="1235" alt="image" src="https://user-images.githubusercontent.com/5927686/206216437-e3061f3d-2794-4b69-ae82-527742af8253.png">
<img width="403" alt="image" src="https://user-images.githubusercontent.com/5927686/206216606-ef05b000-6096-4ec6-8dc8-4d527fab0e72.png">

**Comment by byme8** [comment]
Does it happen when you run the app via CLI?

**Comment by ktomanelli** [comment]
Okay so I now have the repo i sent with your example getting past the bootstrap error and hitting the endpoint which doesnt exist on my end, so thats good. However once I replace the dummy schema with a real one from my hotchocolate server and adjust the query to retrieve actual data, I'm once again getting that bootstrap error.

**Comment by ktomanelli** [comment]
> Does it happen when you run the app via CLI?
> 
> ```
> dotnet run 
> ```

System.InvalidOperationException: Query is not bootstrapped.
   at ZeroQL.GraphQLClient`2.Execute[TVariables,TOperationType,TResult](TVariables variables, Func`3 queryMapper, String queryKey)
   at GraphQLClientRequestExtensions.Execute[TQuery,TResult](IGraphQLClient client, GraphQL`2 request)
   at Program.<Main>$(String[] args) in /Users/kyletomanelli/dev/zeroqlSetup/QLClient/Program.cs:line 13

**Comment by byme8** [comment]
In this case, I have no idea what could go wrong, except something bad happened with your local dotnet setup.

**Comment by byme8** [comment]
@ktomanelli, here is the thread for a very simular issue.
https://youtrack.jetbrains.com/issue/RIDER-82960/Source-Generator-Code-is-not-getting-recognised

They managed to fix it by installing .net 6.0.401

**Comment by be1223** [comment]
**UPDATE**

After trial and error throughout the day, I decided to change my "lambda" syntax to "request" syntax against ZeroQL 3.1.0 and that is now working.

Has there been an issue introduced into the lambda syntax since 3.1.0 was released?

**Comment by byme8** [comment]
When you hover over the `` Query `` method, can you see the graphql preview?

https://user-images.githubusercontent.com/5927686/207651514-3a9733ec-0841-40fb-ac0a-a1135ba42841.mov

**Comment by be1223** [comment]
Yes, using both syntax styles i can hover over the `_client.Query` and it gives me the full graphql preview that i would expect (it is fully correct)

it builds and runs fine, but when it comes to execute the query it throws the "Query is not Bootstrapped" exception, but to reiterate, this is just the lambda syntax - I have just now had a successful execution using the request syntax

**Comment by byme8** [comment]
> Has there been an issue introduced into the lambda syntax since 3.1.0 was released?
I don't think so. It is difficult to say because it has been working fine for me and at least 5+ people I know personally. However, in version 3.2.0-preview.2, there is a change in lambda syntax detection. You can try this version.

**Comment by byme8** [comment]
Can you try to run and send output here
``` bash
dotnet build -v diag > out.txt
```
Maybe there will be something interesting.

**Comment by byme8** [comment]
@be1223 Have you been using ZeroQL before 3.1.0?

**Comment by byme8** [comment]
Before v2.0.0-preview.6 to be precise.

**Comment by be1223** [comment]
@byme8 Not sure what the problem was, but lambda syntax is now also working for me in the same project, chances are i had some error in the syntax.

Regardless, lambda and request syntax are both working for me now!

**Comment by gaffo** [comment]
I too am having this issue intermittently... I have a test repo up at https://github.com/gaffo/CSharpHotChocolateZeroQLIntegrationTestExample with a branch at ZQL-21 on 3.2.0-preview.4 ... but it seems to be a non-deterministic bug. Just changing versions and regenerating seemed to have made it work... There was even a run where one of the mutations was failing but the exact same mutation on a different unit test that ran in the same run worked.

**Comment by byme8** [comment]
I managed to reproduce a similar issue with 3.2.0-preview.3. It happened because of Microsoft.CodeAnalysis.CSharp NuGet package 4.4.0. As far as I understood, it was too new. To use it, you need to have the latest dotnet SDK installed.

In 3.2.0-preview.4 I downgraded Microsoft.CodeAnalysis.CSharp to version 4.0.0(year old version), and now it works as expected. 

@be1223 @ktomanelli Can you try the version 3.2.0-preview.4?

**Comment by byme8** [comment]
@gaffo To clarify. you got an error, changed the version to 3.2.0-preview.4 and now it works as expected?

**Comment by gaffo** [comment]
Actually I had to run the tests multiple times and after a few runs all of them passed without me changing any code. I'm not sure if there was an incremental compile going on in rider or what... One of the mutation tests failed (you can look at them and see they're almost identical) while the other passed, the failure was due to the bootstrapped error.

**Comment by gaffo** [comment]
Okay, I tried it again with 3.2.0-preview.4 on https://github.com/gaffo/CSharpHotChocolateZeroQLIntegrationTestExample/tree/ZQL-21 and it fails with the query issue on all 3 integration tests... with 3.2.0-preview.4 and a clean build it works. So seems to have been identified and fixed in 3.2.0-preview.3 or 3.2.0-preview.4... you can see the diff at https://github.com/gaffo/CSharpHotChocolateZeroQLIntegrationTestExample/commit/7b5867eb2f3ba580816679139047439308547ee7 and it looks like it's entirely contained in the library since the generated code is unchanged.

**Comment by byme8** [comment]
@gaffo, I managed to reproduce the issue on your repository.
Steps are next: 
- create global.json and pin dotnet version to 6.0.101.
- use ZeroQL 3.2.0-preview.2

When you compile the tests project it emits warnings like that:
```
  An instance of analyzer ZeroQL.SourceGenerators.Analyzers.QueryLambdaAnalyzer cannot be created from C:\Users\silin\.nuget\packages\zeroql\3.2.0-preview.2\analyzers\dotnet\cs\ZeroQL.SourceGenerators.dll : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.4.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified..
```
If you run any test it will fail with the `` Query is not bootstrapped`` error.

However, if I use  3.2.0-preview.4 it works fine.

**Comment by byme8** [comment]
> I tried it again with 3.2.0-preview.4
Have you meant 3.2.0-preview.2?

**Comment by gaffo** [comment]
I updated my comment... you are correct .2 fails, .4 works...

Any way to make that warning an error instead it seems that should just fail during build.

**Comment by byme8** [comment]
Issue is fixed in [v3.2.0](https://github.com/byme8/ZeroQL/releases/tag/v3.2.0).
